src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Commands/RejectApprenticeshipUpdates/RejectApprenticeshipUpdatesCommandHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.CommitmentsV2.Authentication;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Data.Extensions;
+using SFA.DAS.CommitmentsV2.Domain.Exceptions;
+using SFA.DAS.CommitmentsV2.Domain.Interfaces;
+using SFA.DAS.CommitmentsV2.Models;
+using SFA.DAS.CommitmentsV2.Shared.Interfaces;
+using SFA.DAS.CommitmentsV2.Types;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.CommitmentsV2.Application.Commands.RejectApprenticeshipUpdates
+{
+    public class RejectApprenticeshipUpdatesCommandHandler : AsyncRequestHandler<RejectApprenticeshipUpdatesCommand>
+    {
+        private readonly Lazy<ProviderCommitmentsDbContext> _dbContext;
+
+        private readonly IAuthenticationService _authenticationService;
+        private readonly ILogger<RejectApprenticeshipUpdatesCommandHandler> _logger;
+
+        public RejectApprenticeshipUpdatesCommandHandler(Lazy<ProviderCommitmentsDbContext> dbContext,
+            IAuthenticationService authenticationService,
+            ILogger<RejectApprenticeshipUpdatesCommandHandler> logger)
+        {
+            _dbContext = dbContext;
+            _authenticationService = authenticationService;
+            _logger = logger;
+        }
+
+        protected override async Task Handle(RejectApprenticeshipUpdatesCommand command, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("RejectApprenticeshipUpdatesCommand received from ApprenticeshipId :" + command.ApprenticeshipId);
+            var party = _authenticationService.GetUserParty();
+            var apprenticeship = await _dbContext.Value.GetApprenticeshipAggregate(command.ApprenticeshipId, cancellationToken);
+            CheckPartyIsValid(party, command, apprenticeship);
+
+            if (apprenticeship.ApprenticeshipUpdate.FirstOrDefault(x => x.Status == ApprenticeshipUpdateStatus.Pending) == null)
+            {
+                throw new InvalidOperationException($"No existing apprenticeship update pending for apprenticeship {command.ApprenticeshipId}");
+            }
+
+
+            apprenticeship.RejectApprenticeshipUpdate(party, command.UserInfo);
+        }
+
+        private void CheckPartyIsValid(Party party, RejectApprenticeshipUpdatesCommand command, Apprenticeship apprenticeship)
+        {
+            if (party != Party.Employer)
+            {
+                throw new DomainException(nameof(party), $"Only employers are allowed to edit the end of completed records - {party} is invalid");
```

Comment:
the term completed records makes me think of the apprenticeship status of completed which is not what this means I don't think?